protocol/query/query.json

```diff
@@ -1,44 +1,44 @@
 {
-	"$schema": "http://json-schema.org/draft-07/schema#",
-	"$id": "https://raw.githubusercontent.com/dedis/student_21_pop/master/protocol/query/query.json",
-	"title": "Match a custom JsonRpc 2.0 query message",
-	"description": "Match a client query",
-	"type": "object",
-	"allOf": [
-		{
-			"maxProperties": 4,
-			"$comment": "Note: can't use \"additionalProperties: false\" due to \"allOf\""
-		},
-		{
-			"properties": {
-				"jsonrpc": {
-					"description": "[String] JsonRpc version",
-					"const": "2.0",
-					"$comment": "should always be \"2.0\""
-				}
-			},
-			"required": [
-				"jsonrpc"
-			]
-		},
-		{
-			"oneOf": [
-				{
-					"$ref": "method/subscribe.json"
-				},
-				{
-					"$ref": "method/unsubscribe.json"
-				},
-				{
-					"$ref": "method/catchup.json"
-				},
-				{
-					"$ref": "method/broadcast.json"
-				},
-				{
-					"$ref": "method/publish.json"
-				}
-			]
-		}
-	]
-}
\ No newline at end of file
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/dedis/student_21_pop/master/protocol/query/query.json",
+    "title": "Match a custom JsonRpc 2.0 query message",
+    "description": "Match a client query",
+    "type": "object",
+    "additionalProperties": false,
```

Comment:
I'd keep this file to a minimum and just reference the children (i.e. the `oneOf` section).

On a side note, `params` is an optional field in JSON-RPC, and while we don't have the case for now, I don't see the reason to make it mandatory at this level.

Reply:
Feel free to reinterpret this in the light of our conversation :-) possibly minimal changes needed